Author: Jamie Strandboge <jamie@canonical.com>
Description: if import of gnomeapplet fails, default to upper panel
 orientation. Inspired by work by Alberto Milone in preparation for
 indicator work. We continue to ship this patch even though we don't ship
 the gnome-panel hamster applet because indicator-hamster currently imports
 applet.py for other things.
Forwarded: not-needed (upstream is porting to gnome3)

Index: hamster-applet-2.91.3+git20110714.9aefd7/src/hamster/applet.py
===================================================================
--- hamster-applet-2.91.3+git20110714.9aefd7.orig/src/hamster/applet.py	2011-08-19 13:32:18.000000000 -0500
+++ hamster-applet-2.91.3+git20110714.9aefd7/src/hamster/applet.py	2011-11-07 11:35:07.000000000 -0600
@@ -26,7 +26,12 @@
 pygtk.require("2.0")
 import gtk, pango
 
-import gnomeapplet
+try:
+    import gnomeapplet
+except ImportError:
+    logging.warning("Could not import gnomeapplet. Defaulting to upper panel")
+    gnomeapplet = None
+
 import gobject
 import dbus, dbus.service, dbus.mainloop.glib
 import locale
@@ -110,9 +115,10 @@
 
         popup_dir = self.get_parent().get_orient()
 
-        orient_vertical = popup_dir in [gnomeapplet.ORIENT_LEFT] or \
-                          popup_dir in [gnomeapplet.ORIENT_RIGHT]
-
+        orient_vertical = False
+        if gnomeapplet != None:
+            orient_vertical = popup_dir in [gnomeapplet.ORIENT_LEFT] or \
+                              popup_dir in [gnomeapplet.ORIENT_RIGHT]
 
         context = self.label.get_pango_context()
         metrics = context.get_metrics(self.label.style.font_desc,
@@ -148,9 +154,9 @@
 
         orient_vertical = True
         new_size = allocation.width
-        if self.popup_dir in [gnomeapplet.ORIENT_LEFT]:
+        if gnomeapplet != None and self.popup_dir in [gnomeapplet.ORIENT_LEFT]:
             new_angle = 270
-        elif self.popup_dir in [gnomeapplet.ORIENT_RIGHT]:
+        elif gnomeapplet != None and self.popup_dir in [gnomeapplet.ORIENT_RIGHT]:
             new_angle = 90
         else:
             new_angle = 0
@@ -453,7 +459,10 @@
 
         self.popup_dir = self.applet.get_orient()
 
-        if self.popup_dir in (gnomeapplet.ORIENT_DOWN, gnomeapplet.ORIENT_UP):
+        if gnomeapplet == None:
+            # Default to upper panel (ORIENT_DOWN)
+            y = y + label.height
+        elif self.popup_dir in (gnomeapplet.ORIENT_DOWN, gnomeapplet.ORIENT_UP):
             if self.popup_dir == gnomeapplet.ORIENT_DOWN:
                 y = y + label.height
             else:
